src/models/storage.ts

```diff
@@ -62,15 +74,17 @@ class EntryStorage {
           try {
             chrome.storage.sync.get((_data: {[hash: string]: OTPStorage}) => {
               for (const hash of Object.keys(_data)) {
-                // decrypt the data to export
-                _data[hash].secret = _data[hash].encrypted ?
-                    encryption.getDecryptedSecret(_data[hash].secret) :
-                    _data[hash].secret;
-                _data[hash].encrypted = false;
-                // we need correct hash
-                if (hash !== _data[hash].hash) {
-                  _data[_data[hash].hash] = _data[hash];
-                  delete _data[hash];
+                if (this.ensureObject(_data, hash)) {
```

Comment:
we can just skip the loop if ensureObject is false:
```
if (!this.ensureObject(_data, hash)) {
    continue;
}
```

Also, `ensureObject` is not a good name here, the name makes me think that the method will fix bad data issue, as what `ensureUniqueIndex` does, but it just returns check result. Suggestion: call it `isValidEntry`.

Reply:
I wouldn't call it `isValidEntry` unless it check if it's an OTPEntry too, I should actually check that too

Reply:
It should be `OTPStorage`, I messed the type. Maybe `isValidOTPStorage`?

Reply:
Or is this method just for checking object, but not for `OTPStorage` checking?

Reply:
It is `OTPStorage`, oops. `isValidEntry` should be fine as long as I make it check everything you suggested, `isValidOTPStorage` is a bit long if it should be called that then `isOTPStorage` should be enough.